### PR TITLE
Convert AbstractArray{ComplexF64} to CuArray{ComplexF32} by default

### LINKED ISSUE #91: cu() behavior for complex floating point numbers
`cu()` is opinionated and automatically converts Float64 to Float32 on the GPU.

It seems like it would make sense for it also to convert ComplexF64 to ComplexF32. 

Could we consider adding this behavior?

### PR DESCRIPTION
**Pull request by p-zubieta**
Fixes #91

### CODE REVIEW

File: src/array.jl [hunk 1]
```diff
@@ -319,6 +319,9 @@ Adapt.adapt_storage(::Float32Adaptor, xs::AbstractArray) =
 Adapt.adapt_storage(::Float32Adaptor, xs::AbstractArray{<:AbstractFloat}) =
   isbits(xs) ? xs : convert(CuArray{Float32}, xs)
 
+Adapt.adapt_storage(::Float32Adaptor, xs::AbstractArray{<:Complex{<:AbstractFloat}}) =
+  isbits(xs) ? xs : convert(CuArray{Float32}, xs)
```
Comment (maleadt) [inline src/array.jl]: Does this work?

```
julia> convert(CuArray{Float32}, ComplexF64[1])
1-element CuArray{Float32,1}:
 1.0

julia> convert(CuArray{ComplexF32}, ComplexF64[1])
1-element CuArray{Complex{Float32},1}:
 1.0f0 + 0.0f0im
```

Comment (p-zubieta) [reply src/array.jl]: Both of those work as expected.

Comment (maleadt) [reply src/array.jl]: What I meant is that you're calling `convert(CuArray{Float32}, xs)` for arguments of eltype `Complex{<:AbstractFloat}`, which results in:

```
julia> cu(ComplexF64[1])
1-element CuArray{Float32,1}:
 1.0

julia> cu(ComplexF64[1+1im])
ERROR: InexactError: Float32(1.0 + 1.0im)
```

Comment (p-zubieta) [reply src/array.jl]: Oh, that was a typo, I had the correct definition in the REPL when I was checking this worked. Should be fixed now.

**Comment by maleadt** [comment]
Huh. Let's try again.
bors try

**Pull request merged**